lib/std/math/test/log1p.zig

```diff
@@ -0,0 +1,103 @@
+const math = @import("../../math.zig");
+const test_utils = @import("../test.zig");
+const Testcase = test_utils.Testcase;
+const runTests = test_utils.runTests;
+const floatFromBits = test_utils.floatFromBits;
+const negInf = test_utils.negInf;
+const nan32 = math.nan_f32;
+const nan64 = math.nan_f64;
+
+const Tc32 = Testcase(math.log1p, "log1p", f32);
+const tc32 = Tc32.init;
+
+const Tc64 = Testcase(math.log1p, "log1p", f64);
+const tc64 = Tc64.init;
+
+// Special-case tests shared between different float sizes, see genTests().
+const special_tests = .{
+    // zig fmt: off
+    .{ 0,         0       },
+    .{-0,        -0       },
+    .{-1,         negInf  },
+    .{ 1,         math.ln2},
+    .{-2,         math.nan},
+    .{ math.inf,  math.inf},
+    .{ negInf,    math.nan},
+    // zig fmt: on
+};
+
+test "math.log1p_32() sanity" {
+    const cases = [_]Tc32{
+        // zig fmt: off
+        tc32(-0x1.0223a0p+3,  nan32        ),
+        tc32( 0x1.161868p+2,  0x1.ad1bdcp+0),
+        tc32(-0x1.0c34b4p+3,  nan32        ),
+        tc32(-0x1.a206f0p+2,  nan32        ),
+        tc32( 0x1.288bbcp+3,  0x1.2a1ab8p+1),
+        tc32( 0x1.52efd0p-1,  0x1.041a4ep-1),
+        tc32(-0x1.a05cc8p-2, -0x1.0b3596p-1),
+        tc32( 0x1.1f9efap-1,  0x1.c88344p-2),
+        tc32( 0x1.8c5db0p-1,  0x1.258a8ep-1),
+        tc32(-0x1.5b86eap-1, -0x1.22b542p+0),
+        // zig fmt: on
+    };
+    try runTests(cases);
+}
+
+test "math.log1p_32() special" {
+    const cases = test_utils.genTests(Tc32, special_tests) ++ test_utils.nanTests(Tc32);
+    try runTests(cases);
+}
+
+test "math.log1p_32() boundary" {
+    const cases = [_]Tc32{
+        // zig fmt: off
+        tc32( 0x1.fffffep+127,  0x1.62e430p+6), // Max input value
+        tc32( 0x1p-149,         0x1p-149     ), // Min positive input value
+        tc32(-0x1p-149,        -0x1p-149     ), // Min negative input value
+        tc32( 0x1p-126,         0x1p-126     ), // First subnormal
+        tc32(-0x1p-126,        -0x1p-126     ), // First negative subnormal
+        tc32(-0x1.fffffep-1,   -0x1.0a2b24p+4), // Last value before result is -inf
+        tc32(-0x1.000002p+0,    nan32        ), // First value where result is nan
+        // zig fmt: on
+    };
+    try runTests(cases);
+}
+
+test "math.log1p_64() sanity" {
+    const cases = [_]Tc64{
+        // zig fmt: off
+        tc64(-0x1.02239f3c6a8f1p+3,  nan64               ),
+        tc64( 0x1.161868e18bc67p+2,  0x1.ad1bdd1e9e686p+0), // Disagrees with GCC in last bit
```

Comment:
If you have stuff like `Disagrees`, always put a `TODO` so you later dont forget.

Reply:
Well, this was explicitly *not* considered as a TODO, rather just a note - this value agrees with Musl (which our implementation is based on) but not with GCC.

Musl seems to say the guarantee is that the error for `log1p()` is always less than 1 ULP (see [here](https://git.musl-libc.org/cgit/musl/tree/src/math/log1p.c#n36)), which leaves room for the last bit being wrong - are you saying this is a violation of IEEE754 and a different implemetation should be used? I'm not sure whether the GCC guarantees are stronger.

Reply:
You probably want this note then near the top level decl.
I checked the musl source and it looks like they ported all their tests to be out of tree into [libc-test](https://wiki.musl-libc.org/libc-test.html).
Their thoroughly tests are also GPLed, but most are MIT.
For example, log1p.h has 550 test cases heavily commented.

If you run `rg 02239f3c6a8f1` in their repo you get alot of matches on `-0x1.02239f3c6a8f1p+3` and a macro like this:
```c
T(RN,   -0x1.02239f3c6a8f1p+3,                     nan,          0x0p+0, INVALID)
```

Should we port this behavior or do you even want to write a helper script to convert that macro stuff?
To me, it sounds simpler to track all upstream test changes then and you dont need to manually port them.